packages/backend/src/server/api/endpoints/hashtags/users.ts

```diff
@@ -9,42 +8,15 @@ export const meta = {
 	tags: ['hashtags', 'users'],
 
 	params: {
-		tag: {
-			validator: $.str,
-		},
-
-		limit: {
-			validator: $.optional.num.range(1, 100),
-			default: 10,
-		},
-
-		sort: {
-			validator: $.str.or([
-				'+follower',
-				'-follower',
-				'+createdAt',
-				'-createdAt',
-				'+updatedAt',
-				'-updatedAt',
-			]),
-		},
-
-		state: {
-			validator: $.optional.str.or([
-				'all',
-				'alive',
-			]),
-			default: 'all',
-		},
-
-		origin: {
-			validator: $.optional.str.or([
-				'combined',
-				'local',
-				'remote',
-			]),
-			default: 'local',
+		type: 'object',
+		properties: {
+			tag: { type: 'string', },
+			limit: { type: 'integer', maximum: 100, default: 10, },
+			sort: { type: 'string', enum: ['+follower', '-follower', '+createdAt', '-createdAt', '+updatedAt', '-updatedAt'], },
```

Comment:
maybe use a pattern here instead?
```suggestion
			sort: { type: 'string', pattern: /^[-+](follower|createdAt|updatedAt)$/, },
```
the code is shorter, but I'm not sure about performance...